### PR TITLE
Fixing Frida ASAN tests on Windows

### PR DESCRIPTION
**Pull request by mkravchik**
This PR fixes libafl_frida ASAN tests on Windows. There are a few things left to fix: the newly added Frida allocator test (on Windows) and adding these tests to the CI on Windows (it looks like they do not run; guidance on how to add them properly would be appreciated).

### CODE REVIEW

**Comment by tokatoka** [comment]
you should make a fuzzer folder named windows_asan_test and there make a Makefile.toml file to run the test

**Comment by mkravchik** [comment]
> you should make a fuzzer folder named windows_asan_test and there make a Makefile.toml file to run the test

So it will be different between windows and linux?

**Comment by tokatoka** [comment]
ok

**Comment by tokatoka** [comment]
wait but i think your log is too redundant

**Comment by mkravchik** [comment]
About your test suggestion - do you know if any other tests are set up like that?

**Comment by tokatoka** [comment]
there's cmplog test in frida_gdi.

**Comment by tokatoka** [comment]
@mkravchik 
I just noticed this...
Could you please remove building test_harness.cpp from the libafl_frida's build script?
It's more appropriate to do this in one of the fuzzer example, not in the main build script

**Comment by s1341** [comment]
@tokatoka I disagree. this is part of the frida unit test suite. not a fuzzer...

File: libafl_frida/src/lib.rs [hunk 1]
```diff
@@ -375,6 +377,14 @@ mod tests {
         executor::FridaInProcessExecutor,
         helper::FridaInstrumentationHelper,
     };
+    #[cfg(unix)]
+    #[global_allocator]
+    static GLOBAL: MiMalloc = MiMalloc;
```
Comment (domenukk) [inline libafl_frida/src/lib.rs:382]: A library should never set the global allocator

**Comment by domenukk** [comment]
In any case, we should only build it when running tests, not everytime

**Pull request merged**